File: hw/medina/u1/HW07ChickenFarmSimulatorWithImput/newChickenPython/view/ChickenFarmSimulator_view.py
import sys
import os
from datetime import date, datetime

sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from model.Chicken import Chicken

def calcular_edad(fecha_nacimiento):
    today = date.today()
    return today.year - fecha_nacimiento.year - ((today.month, today.day) < (fecha_nacimiento.month, fecha_nacimiento.day))

def crear_nueva_gallina(proximo_id):
    print("\n Vamos a crear una nueva gallina")
    name = input("Enter the chicken's name: ")
    color = input("Enter the chicken's color: ")
    is_molting_input = input("Is the chicken molting? (yes/no): ").strip().lower()
    is_molting = is_molting_input == "yes"

    birth_date_str = input("Enter the chicken's birth date (YYYY-MM-DD): ")
    birth_date = datetime.strptime(birth_date_str, "%Y-%m-%d").date()

    age = calcular_edad(birth_date)
    new_chicken = Chicken(proximo_id, name, color, age, is_molting)
    return new_chicken

def main():
    print(" Chicken Farm Simulator\n")

    owner = "Joseph Medina"

    chicken1 = Chicken(1, "Lucy", "White and Brown", 2, False)
    chicken2 = Chicken(2, "Maruja", "White", 1, True)

    gallinas = [chicken1, chicken2]

    for c in gallinas:
        print(f"The chicken is --> {c}")
        print(f"His owner and friend is --> {owner}\n")
        c.do_stuff()
        print("\n---------------------\n")

    while True:
        add_new = input("Do you want to add a new chicken? (yes/no): ").strip().lower()
        if add_new != "yes":
            break

        proximo_id = len(gallinas) + 1
        nueva_gallina = crear_nueva_gallina(proximo_id)
        gallinas.append(nueva_gallina)

        print(f"\nThe new chicken is --> {nueva_gallina}")
        print(f"His owner and friend is --> {owner}\n")
        nueva_gallina.do_stuff()
        print("\n---------------------\n")

    print("\n--- End of Chicken Farm Simulation ---")

if __name__ == "__main__":
    main()

File: hw/medina/u1/hW07ChickenFarmSimulatorWithImput/newChickenJS/view/ChickenFarmSimulator_view.js
const readline = require("readline");
const { Chicken } = require("../model/Chicken");

function ask(question) {
  return new Promise((resolve) => {
    rl.question(question, (answer) => resolve(answer));
  });
}

function calculateAge(birthDate) {
  const today = new Date();
  let age = today.getFullYear() - birthDate.getFullYear();
  const m = today.getMonth() - birthDate.getMonth();
  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
    age--;
  }
  return age;
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const chickens = [
  new Chicken(1, "Lucy", "White and Brown", 2, false),
  new Chicken(2, "Maruja", "White", 1, true),
];

async function main() {
  console.log("This is my Chicken Farm Simulator\n");

  const owner = "Joseph Medina";

  for (const chicken of chickens) {
    console.log("The chicken is --> " + chicken.toString());
    console.log("His owner and friend is --> " + owner);
    chicken.doStuff();
    console.log("------------------------------\n");
  }

  while (true) {
    const addMore = (await ask("Do you want to add a new chicken? (yes/no): ")).toLowerCase();
    if (addMore !== "yes") break;

    const name = await ask("Enter chicken name: ");
    const color = await ask("Enter chicken color: ");
    const isMoltingAnswer = await ask("Is it molting? (yes/no): ");
    const isMolting = isMoltingAnswer.toLowerCase() === "yes";
    const birthStr = await ask("Enter birth date (YYYY-MM-DD): ");
    const birthDate = new Date(birthStr);
    const age = calculateAge(birthDate);

    const id = chickens.length + 1;
    const newChicken = new Chicken(id, name, color, age, isMolting);
    chickens.push(newChicken);

    console.log("\nThe chicken is --> " + newChicken.toString());
    console.log("His owner and friend is --> " + owner);
    newChicken.doStuff();
    console.log("------------------------------\n");
  }

  console.log("Simulation ended.");
  rl.close();
}

main();
